src/cli/mod.rs

```diff
@@ -0,0 +1,241 @@
+    limit: Option<usize>,
+    // /// Reduction count used for proofs (defaults to 10)
+    // #[clap(long, value_parser)]
+    // rc: Option<usize>,
+
+    // /// Flag to prove the last evaluation
+    // #[arg(long)]
+    // prove: bool,
+}
+
+#[derive(Parser, Debug)]
+struct LoadCli {
+    #[clap(value_parser)]
+    lurk_file: PathBuf,
+
+    #[clap(long, value_parser)]
+    zstore: Option<PathBuf>,
+
+    #[clap(long, value_parser)]
+    limit: Option<usize>,
+    // #[arg(long)]
+    // prove: bool,
+
+    // #[clap(long, value_parser)]
+    // rc: Option<usize>,
+}
+
+impl LoadArgs {
+    pub fn into_cli(self) -> LoadCli {
+        LoadCli {
+            lurk_file: self.lurk_file,
+            zstore: self.zstore,
+            limit: self.limit,
+            // prove: self.prove,
+            // rc: self.rc,
+        }
+    }
+}
+
+#[derive(Args, Debug)]
+struct ReplArgs {
+    /// ZStore to be preloaded before entering the REPL (and loading a file)
+    #[clap(long, value_parser)]
+    zstore: Option<PathBuf>,
+
+    /// Optional file to be loaded before entering the REPL
+    #[clap(long, value_parser)]
+    load: Option<PathBuf>,
+
+    /// Maximum number of iterations allowed (defaults to 100_000_000)
+    #[clap(long, value_parser)]
+    limit: Option<usize>,
+    // /// Reduction count used for proofs (defaults to 10)
+    // #[clap(long, value_parser)]
+    // rc: Option<usize>,
+}
+
+#[derive(Parser, Debug)]
+struct ReplCli {
+    #[clap(long, value_parser)]
+    load: Option<PathBuf>,
+
+    #[clap(long, value_parser)]
+    zstore: Option<PathBuf>,
+
+    #[clap(long, value_parser)]
+    limit: Option<usize>,
+    // #[clap(long, value_parser)]
+    // rc: Option<usize>,
+}
+
+impl ReplArgs {
+    pub fn into_cli(self) -> ReplCli {
+        ReplCli {
+            load: self.load,
+            zstore: self.zstore,
+            limit: self.limit,
+            // rc: self.rc,
+        }
+    }
+}
+
+fn get_field() -> Result<LanguageField> {
+    if let Ok(lurk_field) = std::env::var("LURK_FIELD") {
+        match lurk_field.to_lowercase().as_str() {
+            "bls12-381" => Ok(LanguageField::BLS12_381),
+            "pallas" => Ok(LanguageField::Pallas),
+            "vesta" => Ok(LanguageField::Vesta),
+            _ => bail!("Field not supported: {lurk_field}"),
+        }
+    } else {
+        Ok(DEFAULT_FIELD)
+    }
+}
+
+fn get_store<F: LurkField + for<'a> serde::de::Deserialize<'a>>(
+    zstore_path: &Option<PathBuf>,
+) -> Result<Store<F>> {
+    match zstore_path {
+        None => Ok(Store::default()),
+        Some(zstore_path) => {
+            info!("Reading ZStore bytes");
+            let bytes = fs::read(zstore_path)?;
+            info!("Deserializing ZStore bytes into ZData");
+            let zdata = ZData::from_bytes(&bytes)?;
+            info!("Deserializing ZData into ZStore");
+            let zstore: ZStore<F> = from_z_data(&zdata)?;
+            info!("Converting ZStore into Store");
```

Comment:
This is a bit verbose, no? 

I would have expected an error log in case the store reads would have not worked, **in the case where we will recover from this operation and proceed with an empty store**, something like `frobnicate().tap_err(|e| log::error!("Failed to frobnicate: {e}"))`. 

Of course, if we're going to fail and give up on the operation, then that failure would make the error explicit to the user, so we should not need the log as much.

Reply:
Yeah, while I was implementing it, I realized that it's actually important to just fail rather than proceeding with an empty store just to (potentially) fail later due to something the user expects to be in the store but it's not there. That would save resources (including time) for the user.

So, should I just remove the logs in this case?

Reply:
Yes, that seems sensible. Any panic should show its embedded error clearly.